src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarm.java

```diff
@@ -488,7 +488,31 @@ public ListSuitesResult listSuites(String runArn) {
         return api.listSuites(request);
     }
 
+    /*
+     * get number of Unmetered devices based on upload type
+     */
+    public int getUnmeteredDevices(String os) {
+        if (os.equalsIgnoreCase("Android")) {
+            return getAccountSettings().getUnmeteredDevices().get("ANDROID");
+        } else if (os.equalsIgnoreCase("IOS")) {
+            return getAccountSettings().getUnmeteredDevices().get("IOS");
+        } else {
+            return 0;
+        }
+    }
+
+    public String getOs(String appArtifact) {
+        if (appArtifact.toLowerCase().endsWith("apk")) {
+            return "Android";
+        } else {
+            return "IOS";
+        }
+    }
 
+    public AccountSettings getAccountSettings() {
+        GetAccountSettingsRequest request = new GetAccountSettingsRequest();
+        return api.getAccountSettings(request).getAccountSettings();
```

Comment:
getAccountSettings() can throw a NotFoundException is the account isn't configured for unmetered usage.